code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductCatalogNotificationBuilder.kt

```diff
@@ -0,0 +1,335 @@
+     */
+    private fun RemoteViews.setCtaButtonColor(
+        containerViewId: Int,
+        buttonColor: String?
+    ) {
+        // get custom color from hex string and set it the cta button
+        if (buttonColor.isNullOrEmpty()) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "Empty cta button color hex string found, custom color will not be applied to the cta button."
+            )
+            return
+        }
+        setElementColor(
+            containerViewId,
+            "#$buttonColor",
+            PushTemplateConstants.MethodNames.SET_BACKGROUND_COLOR,
+            PushTemplateConstants.FriendlyViewNames.CTA_BUTTON
+        )
+    }
+
+    /**
+     * Creates a pending intent for a thumbnail interaction in a product catalog notification.
+     *
+     * @param context the application [Context]
+     * @param broadcastReceiverClass the [Class] of the broadcast receiver to set in the created pending intent
+     * @param channelId [String] containing the notification channel ID
+     * @param pushTemplate the [ProductCatalogPushTemplate] object containing the product catalog push template data
+     * @param currentIndex [Int] containing the index of the current product in the catalog
+     * @return the created thumbnail interaction [PendingIntent]
+     */
+    private fun createThumbnailInteractionPendingIntent(
+        context: Context,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?,
+        channelId: String,
+        pushTemplate: ProductCatalogPushTemplate,
+        currentIndex: Int
+    ): PendingIntent? {
+        if (broadcastReceiverClass == null) {
+            return null
+        }
+
+        Log.trace(
+            LOG_TAG,
+            SELF_TAG,
+            "Creating a thumbnail interaction pending intent for thumbnail at index $currentIndex"
+        )
+
+        val thumbnailClickIntent = AEPPushNotificationBuilder.createIntent(intentActionList[currentIndex], pushTemplate).apply {
+            setClass(context.applicationContext, broadcastReceiverClass)
+            putExtra(PushTemplateConstants.PushPayloadKeys.CHANNEL_ID, channelId)
+            putExtra(PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_TEXT, pushTemplate.ctaButtonText)
+            putExtra(PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_COLOR, pushTemplate.ctaButtonColor)
+            putExtra(PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_URI, pushTemplate.ctaButtonUri)
+            putExtra(PushTemplateConstants.PushPayloadKeys.CATALOG_LAYOUT, pushTemplate.displayLayout)
+            putExtra(PushTemplateConstants.PushPayloadKeys.CATALOG_ITEMS, pushTemplate.rawCatalogItems)
+            putExtra(PushTemplateIntentConstants.IntentKeys.CATALOG_ITEM_INDEX, currentIndex.toString())
+        }
+
+        return PendingIntent.getBroadcast(
+            context,
+            0,
```

Comment:
I think we need to set different requestCodes for each thumbnail here, else Android system will consider the PendingIntent for all the thumbnails as the same and it will reuse the first one

Reply:
Oh just saw that the intent action is different for each thumbnail, so this isn't needed